From 1d41934c78b0a8f5fa6a80a17bb5f5c2daf504c8 Mon Sep 17 00:00:00 2001
From: Luca Weiss <luca@z3ntu.xyz>
Date: Mon, 29 Nov 2021 21:16:20 +0100
Subject: [PATCH 09/10] Add possibility to force kms device

Useful on PinePhone as it has two graphics devices, one for gpu and one
for display.
---
 src/platforms/mesa/server/display_helpers.cpp | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/src/platforms/mesa/server/display_helpers.cpp b/src/platforms/mesa/server/display_helpers.cpp
index b357f6c67b..5b8eb2b3e1 100644
--- a/src/platforms/mesa/server/display_helpers.cpp
+++ b/src/platforms/mesa/server/display_helpers.cpp
@@ -57,9 +57,15 @@ mgmh::DRMHelper::open_all_devices(
 {
     int error = ENODEV; //Default error is "there are no DRM devices"
 
+    char* card_to_use = getenv("MIR_MESA_KMS_USE_DRM_DEVICE");
+
     mir::udev::Enumerator devices(udev);
     devices.match_subsystem("drm");
-    devices.match_sysname("card[0-9]");
+    if(card_to_use == nullptr) {
+        devices.match_sysname("card[0-9]");
+    } else {
+        devices.match_sysname(card_to_use);
+    }
 
     devices.scan_devices();
 
-- 
2.34.1

